dotnet/samples/KernelSyntaxExamples/Example99_GeminiEmbeddingGeneration.cs

```diff
@@ -0,0 +1,301 @@
+        });
+
+        this.WriteLine();
+
+        /////////////////////////////////////////////////////////////////////////////////////////////////////
+        // PART 3: Recall similar ideas with semantic search
+        //
+        // Uses AI Embeddings for fuzzy lookup of memories based on intent, rather than a specific key.
+        /////////////////////////////////////////////////////////////////////////////////////////////////////
+
+        WriteLine("== PART 3: Recall (similarity search) with AI Embeddings ==");
+
+        WriteLine("== PART 3a: Recall (similarity search) with ISemanticTextMemory ==");
+        WriteLine("Ask: live in Seattle?");
+
+        await foreach (var answer in textMemory.SearchAsync(
+                           collection: MemoryCollectionName,
+                           query: "live in Seattle?",
+                           limit: 2,
+                           minRelevanceScore: 0.79,
+                           withEmbeddings: true))
+        {
+            WriteLine($"Answer: {answer.Metadata.Text}");
+        }
+
+        /* Possible output:
+         Answer: I've been living in Seattle since 2005
+        */
+
+        WriteLine("== PART 3b: Recall (similarity search) with Kernel and TextMemoryPlugin 'Recall' function ==");
+        WriteLine("Ask: my family is from?");
+
+        var result = await kernel.InvokeAsync(memoryPlugin["Recall"], new()
+        {
+            [TextMemoryPlugin.InputParam] = "Ask: my family is from?",
+            [TextMemoryPlugin.CollectionParam] = MemoryCollectionName,
+            [TextMemoryPlugin.LimitParam] = "2",
+            [TextMemoryPlugin.RelevanceParam] = "0.79",
+        });
+
+        WriteLine($"Answer: {result.GetValue<string>()}");
+        WriteLine();
+
+        /* Possible output:
+         Answer: ["My family is from New York"]
+        */
+
+        /////////////////////////////////////////////////////////////////////////////////////////////////////
+        // PART 4: TextMemoryPlugin Recall in a Prompt Function
+        //
+        // Looks up related memories when rendering a prompt template, then sends the rendered prompt to
+        // the text generation model to answer a natural language query.
+        /////////////////////////////////////////////////////////////////////////////////////////////////////
+
+        WriteLine("== PART 4: Using TextMemoryPlugin 'Recall' function in a Prompt Function ==");
+
+        // Build a prompt function that uses memory to find facts
+        const string RecallFunctionDefinition = @"
+Consider only the facts below when answering questions:
+
+BEGIN FACTS
+About me: {{recall 'live in Seattle?'}}
+About me: {{recall 'my family is from?'}}
+END FACTS
+
+Question: {{$input}}
+
+Answer:
+";
+
+        var aboutMeOracle = kernel.CreateFunctionFromPrompt(RecallFunctionDefinition, new GeminiPromptExecutionSettings() { MaxTokens = 1000 });
+
+        result = await kernel.InvokeAsync(aboutMeOracle, new()
```

Comment:
I imagine this was just copied from some existing sample, but it'd be nice if we strived to make these examples as simple as possible, e.g. if the sample isn't about creating a KernelFunction and InvokePromptAsync would otherwise suffice, just use InvokePromptAsync.

Reply:
done